project_key/models/project_project.py

```diff
@@ -130,6 +130,12 @@ def _prepare_sequence_data(self, init=True):
     def get_next_task_key(self):
         return self.sudo().task_key_sequence_id.next_by_id()
```

Comment:
Shouldn't this instead check if returned key is valid and generate new ones until proper is found?

Reply:
@agajic-modoolar I still see no reason if I've created a task A-200 to enforce next in sequence to be A-201 instead of "next unused number"

Reply:
> Shouldn't this instead check if returned key is valid and generate new ones until proper is found?

Interesting approach but the idea here was to have a "straight line" of task numbers as "jumping" numbers would only confuse people. Imagine you have tasks from "A001" to "A010" and someone deletes task "A003" as that task was not needed anymore. After some time a new task is created and instead of assigning key "A011" you get "A003" again and then the person who deleted "A003" see it and deletes it again thinking it's a bug in the system or something ....  I don't think that a good approach, so that's why we are always using the next sequence which is only natural to me.

Reply:
That's not how it would work. Assume that:
 1) sequence is at 001,
 2) you create 5 more (when task key is not set)
 3) sequence becomes at 005, 
 4) then you import 5 more tasks, with manual names set to "A007-A011", sequence generator was not used since names were set
 5) next time when you create task as usual, it would propose to use A006 and it's still "unused" 
 6) next time when you create task as usual, it would propose to use A007 and yet it's used, so next one 008...011 until 012 would be generated, and sequence would get updated.

I think these are just two approaches that would be good to have as configurable settings - as this is a behaviour-breaking change.

Reply:
Um, if I understand correctly you are afraid that the sequence "A006" will be generated two times?

If, so, i can assure you that it won't happen because, after you import "A007-A011" the next sequence number will be set to "12" (please check line 53) and then when you next time normally create task it will get key "A012".

Reply:
We're not on the same page right now, unfortunately. I understand that it would be set to "A012" in the end. I'm just not sure if this is the desired default approach, and if it should be done as a wizard with import options like "regenerate keys" vs "import keys" vs "import keys + shift sequence"

Reply:
Maybe I may even address that myself in v12